pvanalytics/features/clipping.py

```diff
@@ -223,3 +226,110 @@ def threshold(ac_power, slope_max=0.0035, power_min=0.75,
+    criteria:
+
+    - the normalized value is greater than `clip_min`
+    - the forward or backward difference is less than `derivative_max`
+    - the value is at least `daily_fraction_min` of the maximum value
+      on the same day
+
+    A value that meets the three criteria above must also be part of a
+    sequence of more than `length_min` consecutive values that also meet
+    the clipping criteria.
+
+    On each day, the values that are flagged as clipped according the
+    these criteria are used to identify a daily clipping threshold. The
+    daily threshold is the minimum of the clipped values minus `margin`.
+    Any value on that day that is greater than or equal to the threshold
+    is flagged as clipped and the resulting mask is returned.
+
+    Parameters
+    ----------
+    power_ac : Series
+        AC Power.
+    clip_min : float, default 0.8
+        After normalization a clipped value must be greater than `clip_min`
+    daily_fraction_min : float, default 0.9
+        A clipped value must be greater than `daily_fraction_min` times the
+        maximum value on the same day.
+    length_min : int, default 2
+        Minimum number of sequential values that satisfy clipping criteria.
+    margin : float, default 0.01
+        Values within +/- `margin` of the daily clipping threshold are flagged
+        as clipped.
+    freq_minutes : float, optional
+        Timestamp spacing in minutes. If not specified timestamp spacing
+        will be inferred from the index of `power_ac`.
+    derivative_max : float, optional
+        Values with a derivative less than `derivative_max` are
+        potentially clipped. If not specified the value is based on
+        the timestamp spacing in `power_ac` according to the following
+        formula:
+
+        .. math::
+
+           0.00005 * f_m + 0.0009
+
+        where :math:`f_m` is the timestamp spacing in minutes.
+
+    Returns
+    -------
+    Series
+        Boolean series with True for values that appear to be clipped.
+    """
+    if freq_minutes is None:
+        freq_minutes = _estimate_freq_minutes(power_ac.index)
+    if derivative_max is None:
+        # Experimentally derived formula from the PVFleets project
+        derivative_max = (0.00005 * freq_minutes) + 0.0009
+    power_ac = power_ac.copy()
+    power_ac.loc[power_ac < 0] = 0
+    power_normalized = _normalize.min_max(power_ac)
+    forward_diff = abs(power_normalized - power_normalized.shift(-1))
+    backward_diff = abs(power_normalized - power_normalized.shift(1))
+    daily_max = _group.by_day(power_normalized).transform('max')
+    fraction_of_max = power_normalized / daily_max
+    candidate_clipping = ((power_normalized >= clip_min)
+                          & ((forward_diff <= derivative_max)
+                             | (backward_diff <= derivative_max))
+                          & (fraction_of_max >= daily_fraction_min))
```

Comment:
Maybe to discuss with Matt: why both `>= clip_min` and `>= daily_fraction_min`?

Reply:
This is in flux. It is likely that the global `clip_min` parameter is not needed and will be removed.